src/core/qgsvectorlayer.cpp

```diff
@@ -4286,6 +4286,15 @@ QVariant QgsVectorLayer::aggregate( QgsAggregateCalculator::Aggregate aggregate,
   if ( ok )
     *ok = false;
 
+  //make a copy or make the param not const
+  QgsAggregateCalculator::AggregateParameters paramCopy = QgsAggregateCalculator::AggregateParameters( parameters );
+  if ( paramCopy.filter.contains( "var('symbol_expression')" ) )
+  {
+    QString symbEx = QgsExpression::replaceExpressionText( "[% ( @symbol_expression ) %]", context );
```

Comment:
@nyalldawson If you think that users understand that they need to add 'eval(' to have this variable as an expression and not as text in the query this may be removed as I understand this may be the most contentious part of this PR.